vis/optimizer.py

```diff
@@ -93,7 +93,8 @@ def _get_seed_input(self, seed_input):
             seed_input = np.expand_dims(seed_input, 0)
 
         # Only possible if channel idx is out of place.
-        if seed_input.shape != desired_shape:
+        if seed_input.shape[-1] != desired_shape[-1] and \
+           seed_input.shape[1] != desired_shape[1]:
```

Comment:
is this to handle the theano case? if so, shouldnt the conditional be 'or'?

Reply:
Yes, it is to handle the Theano case. `and` is correct here. Let have a case where the desired shape is `[1, None, None, 3]` and seed input is [1, 299, 299, 3]. In this case, both representations are channel last and condition will be false since the first part is `True` and the second is `False`. The other case is when the desired shape is `[1, 3, None, None]` and seed input is [1, 299, 299, 3]. In this case, both parts of the condition are `True` and axis shuffling is required.